unified_planning/model/state.py

```diff
@@ -159,3 +160,102 @@ def make_child(
             return UPState(complete_values)
         # Otherwise just return a new UPState with self as ancestor
         return UPState(updated_values, self)
+
+class SparseState(State):
+    """
+    State representation only storing non-default values.
+
+    Has a big performance advantage over UPState in that it does not require explicitly listing
+    false boolean fluents. For this reason also does not try to implement the ancestore trick.
+    """
+
+    def __init__(
+        self,
+        values: Dict["up.model.FNode", "up.model.FNode"],
+        fluent_set: FluentsSetMixin
+    ):
+        """
+        Creates a new `SparseState` where values are those that possibly differ from
+        the defaults specified in the given fluent_set.
+        """
+        self._fluent_set = fluent_set
+
+        self._values = {}
+        for fluent, value in values.items():
+            if not fluent.is_fluent_exp():
+                raise UPValueError(
+                    f"The fluent '{fluent}' is not a fluent expression, but a '{fluent.node_type.name}'"
+                )
+            if not value.is_constant():
+                raise UPValueError(
+                    f"The value '{value}' assigned to the fluent '{fluent}' is not a constant, but a '{value.node_type.name}'"
+                )
+            if self._is_nondefault(fluent,value):
+                self._values[fluent] = value
+
+        self._hash: Optional[int] = None
+
+    def _is_nondefault(self, fluent: "up.model.FNode", value: "up.model.FNode"):
+        return (fluent.fluent() not in self._fluent_set.fluents_defaults or
+                self._fluent_set.fluents_defaults[fluent.fluent()] != value or
+                fluent.type.is_bool_type() and value.is_true())
```

Comment:
```suggestion
                self._fluent_set.fluents_defaults[fluent.fluent()] != value)
```

if `true` is the correct default for the fluent, it is already in the `fluents_defaults` dict